internal/runner/operation.go

```diff
@@ -118,6 +118,23 @@ func (r *Runner) executeJob(
 		return nil, err
 	}
 
+	// Here we'll load our values from the server/UI, as well as VCS.
+	// The order values are added to our final pbVars slice in the order
+	// of precedence
+	pbVars := resp.Project.GetVariables()
+	vcsVars, diags := variables.LoadVCSFiles(wd)
+	if diags.HasErrors() {
+		return nil, diags
+	}
+	pbVars = append(pbVars, vcsVars...)
+	pbVars = append(pbVars, job.Variables...)
```

Comment:
I'd reorder this a bit to:

```go
	vcsVars, diags := variables.LoadVCSFiles(wd)
	if diags.HasErrors() {
		return nil, diags
	}

	pbVars := resp.Project.GetVariables()
	pbVars = append(pbVars, vcsVars...)
	pbVars = append(pbVars, job.Variables...)
```

This is a nit pick but I find it cleaner to bunch together the accumulation if possible.

Reply:
I like it, done